include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -231,7 +342,26 @@ class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 			return true;
 		}
 
+		inline size_t getTotalMutableSamplerCount() const { return m_mutableSamplerCount; }
+		inline size_t getTotalDescriptorCount(const E_DESCRIPTOR_TYPE type) const { return m_descriptorCount[type]; }
+
 		core::SRange<const SBinding> getBindings() const { return {m_bindings->data(), m_bindings->data()+m_bindings->size()}; }
+
+		inline uint32_t getBindingCount(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].count; }
+		inline uint32_t* getBindingStorage(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].bindings.get(); }
+		inline uint32_t* getBindingOffsetStorage(const E_DESCRIPTOR_TYPE type) const { return m_descriptorRedirects[type].offsets; }
```

Comment:
why are these mutable pointer returning functions public!?

Reply:
ok I see they're used in the asset converter to make it nice.

My only request is that you make the pointers returned `const uint32_t*`

Reply:
hmmm actually this is bad design cause passing `uint32_t` looses all semantics, just do a 
```cpp
const SBindingRedirect& getBindingRedirects(const E_DESCRIPTOR_TYPE type);
const SBindingRedirect& getSamplerRedirects();
```

then you don't need anything from line 350 to line 355

Reply:
hmm I think you might need a `SBindingRedirect` once for mutable and once for immutable samplers

Reply:
> hmmm actually this is bad design cause passing `uint32_t` looses all semantics, just do a
> 
> ```c++
> const SBindingRedirect& getBindingRedirects(const E_DESCRIPTOR_TYPE type) const;
> const SBindingRedirect& getSamplerRedirects() const;
> ```
> 
> then you don't need anything from line 350 to line 355
> 
> P.S. obviously `ICPUDescriptorSet` would need the non-`const` reference returning variants subject to asserts about immutability, etc.

My main aim behind this kind of API was not to "reveal" to the user, how descriptor bindings are stored in `IDescriptorSetLayout`. So I added getters and kept the logic of retrieving data from the redirects in `IDescriptorLayout` only. With this, if we wanted to change the way how binding data was stored in the future we could easily do that without disturbing the rest of the engine, basically encapsulation.
But no matter now, I'm returning a `SBindingRedirect&`.

Though, I do agree with your type safety point. I should've returned `const storage_offset_t*` instead of a simple `uint32_t*`

Reply:
Resolved.